ssawg_trending_scraper.py

```diff
@@ -0,0 +1,376 @@
+        for offset in range(4, 0, -1):
+            year = (datetime.now()).year
+            url = (f'{base_url}{page}{year}/Q{offset}/index.html') # creates the temporary url; will later need '/index.html' removed
+            if page == "periscope_drift_reports/": # this page requires a username/password
+                page_request = requests.get(url, auth=(user, password))
+            else:
+                page_request = requests.get(url)
+            if page_request.status_code < 300: # verify successful page
+                soup = BeautifulSoup(page_request.text, "lxml")  
+                temp_tables = soup.find_all('table') # locate table with quarterly dates
+                start_time = temp_tables[1].text[12:34] # quarterly start date
+                stop_time = temp_tables[1].text[34:] # quarterly stop date
+                stop_minus_start = Chandra.Time.date2secs(stop_time) - Chandra.Time.date2secs(start_time)
+                halfway = Chandra.Time.date2secs(start_time) + stop_minus_start
+                if Chandra.Time.date2secs(datetime.now()) > halfway: # if 50% through quarter
+                    return (f'{base_url}{page}{year}/Q{offset}/')
+                elif offset == 1: # if it's the first quarter of the year
+                    return (f'{base_url}{page}{year-1}/Q4/') # switch to fourth quarter of previous year
+                else:
+                    return (f'{base_url}{page}{year}/Q{offset-1}/') # try previous quarter
+            else:
+                continue
+    elif page ==  "perigee_health_plots/": # perform unique checks on this page
+        for offset in range(0, -31, -1): # looks back one month
+            now = datetime.now() + timedelta(days=(offset))
+            if now.day > 15: # if roughly halfway through the month
+                year = now.year
+                month = now.month
+                url = (f'{base_url}{page}SUMMARY_DATA/{year}-M{month:02}/') # provides URL if halfway through month
+                response = requests.get(url)
+                if response.status_code < 300:
+                    return url
+                else:
+                    continue
+            else: # if not halfway through the month, get previous month
+                now = datetime.now() + timedelta(days=-16) # offset of -16 days will get previous month
+                year = now.year
+                month = now.month
+                url = (f'{base_url}{page}SUMMARY_DATA/{year}-M{month:02}/')
+                response = requests.get(url)
+                if response.status_code < 300:
+                    return url
+                else:
+                    continue
+    else:
+        return base_url + page
+
+# -----------------------------
+# GET VARIOUS HTML INFORMATION
+# -----------------------------
+
+def get_images(soup,url):
+    images = {}
+    for img in soup.find_all('img'):
+        if img['src'][-3:] == "png" or img['src'][-3:] == "gif": # look for all pngs and gifs
```

Comment:
Use e.g. `img['src'].endswith('.png')`

Reply:
Whoa! I didn't even know that existed. Would've saved me a lot of trouble early on. Thanks!